pkg/config/runconfig.go

```diff
@@ -173,6 +173,17 @@ MAINLOOP:
 			continue MAINLOOP
 		}
 
+		// With the snapshot updated, get a list of
+		// kubecontrollersconfigurations so we can watch on its resource
+		// version.
+		kccList, err := client.List(ctx, options.ListOptions{Name: "default"})
```

Comment:
I'm not convinced about this. I don't understand why this is needed. 

We should be able to use the resource version returned by the Update() to continue watching - if that's not working I think it's probably a bug somewhere else in our watch logic that we're hiding with this change. Unless I'm missing something...

Reply:
Actually, I think you are right - the Update() above is on a particular item and so it won't necessarily have the same RV as the List result, which is the one we need for Watch(). I think this is good.